### ISSUE TITLE
Downloaded hdf files

### ISSUE THREAD

**Issue by RajasBharambe** [fraymio/modis-tools#36]
**Describe the bug**
I downloaded the hdf files using modis-tools, but unable to open the file: It says File Signature not found
when I browse through some solutions it says, maybe the file is corrupted.

**Your Environment:**
- `modis_tools` version used:
-  Other dependencies and packages installed in environment you are using (e.g. output of `pip freeze` or `conda list` or `poetry show`):

**Comment by cearlefraym** [comment]
Hi @RajasBharambe , can you share the code you used to download the files (omitting your login credentials) so that can try and replicated on my side.

**Comment by RajasBharambe** [comment]
_Sure, here's the code:_

from modis_tools.auth import ModisSession
from modis_tools.resources import CollectionApi, GranuleApi
from modis_tools.granule_handler import GranuleHandler

username = ""
password = ""

session = ModisSession(username=username, password=password)

collection_client = CollectionApi(session=session)
collections = collection_client.query(short_name="MOD11A1", version="061")

granule_client = GranuleApi.from_collection(collections[0], session=session)

india_bbox = [71.00, 29.00, 86.50, 18.00]
granules = granule_client.query(start_date="", end_date="", bounding_box=india_bbox)

file_paths = GranuleHandler.download_from_granules(granules, session, ext = ("hdf", "h5", "nc", "xml"))

GranuleHandler.download_from_granules(granules, session=session, path="")

**Comment by RajasBharambe** [comment]
Sure, i want to download the data from 1990 to 2022

**Comment by cearlefraym** [comment]
@RajasBharambe, I'm not able to recreate that error on my end using a subset of years for that product. Just a few things that might help you troubleshoot further: 
The bounding box you provided looks incorrect. The box should take the form [xmin, ymin, xmax, ymax]. You can try this box for India to see if that changes anything for you:
```
india_bbox = [68.17822, 6.755953, 97.41529, 35.67455]
```
Querying 23 years of data at a time raises some concerns as well. I recommend querying in smaller batches to ensure you have sufficient storage, bandwidth, etc. (I suspect this will reach terrabytes of data) and to ensure the data quality in older time frames works for your use case. 

If you still get this error, please provide any changes to the code you are using as well.

**Comment by RajasBharambe** [comment]
Ohk **@cearlefraym** , I will try it, thanks for your suggestion.

**Comment by cearlefraym** [comment]
Any update on this @RajasBharambe? If this is resolved I can close out the issue

**Comment by RajasBharambe** [comment]
No, its not. Im getting the same error after trying.

**Comment by RajasBharambe** [comment]
Sure, the code is as follows:


```
from modis_tools.auth import ModisSession
from modis_tools.resources import CollectionApi, GranuleApi
from modis_tools.granule_handler import GranuleHandler

username =
password =

# Authenticate a session
session = ModisSession(username=username, password=password)

# Query the MODIS catalog for collections
collection_client = CollectionApi(session=session)
collections = collection_client.query(short_name="MOD11A1", version="061")

# Query the selected collection for granules
granule_client = GranuleApi.from_collection(collections[0], session=session)

# Filter the selected granules via spatial and temporal parameters
india_bbox = [55.00, 5.00, 105.00, 45.00]
india_granules = granule_client.query(start_date="2020-01-01", end_date="2023-01-01", bounding_box=india_bbox)

# Download the granules
GranuleHandler.download_from_granules(india_granules, session)
```

**Comment by RajasBharambe** [comment]
Sir, please tell me what should I do ?

**Comment by cearlefraym** [comment]
The code you posted above is working for me, although it takes a while to start up as you are querying 3 years of data which is over 31000 files.
![image](https://github.com/fraymio/modis-tools/assets/97044512/3267960e-eda3-4d7e-ac5f-e774325305ec)

**Comment by RajasBharambe** [comment]
Yes, I tried doing all the possible ways you mentioned, but same error is occuring. 


![image](https://github.com/fraymio/modis-tools/assets/89535670/07bb4a76-e21d-47c6-95b9-894ad30ea78a)

**Comment by cearlefraym** [comment]
Can you send the full stack trace?

**Comment by cearlefraym** [comment]
Are you still getting this error? And if so, can you please provide a stack trace along with your code to help troubleshoot further?
